### DISCUSSION TITLE
Is there a way to draw pixel arrays dynamically on Image control in MAUI?

### DISCUSSION THREAD

**Discussion by evhenii-komar** [dotnet/maui#7644] [Q&A]
Hi! For every frame, I have a 4-bytes BGRA pixel array to update my Image control. However, I can't find a way to update it in MAUI frequently. Tens times per second.
In WinUI3 I use an Image control with WriteableBitmap as a source. I update WriteableBitmap.PixelBuffer as a stream for every frame and it works fine. WinUI3 pseudo-code is:
```
 private Image Image;
 private WriteableBitmap bitmap = new WriteableBitmap(width, height);
 ...
 Image.Source = bitmap;
 ...
 var stream = bitmap.PixelBuffer.AsStream();
 byte[] pixelsBytes = (byte[])parameters.Pixels; // every 4 bytes is a BGRA pixel
 stream.Seek(0, SeekOrigin.Begin);
 stream.Write(pixelsBytes, 0, pixelsBytes.Length);
 bitmap.Invalidate();
```
In MAUI I tried this approach:
Xaml:
` <Image x:Name="MyImage" HeightRequest="200"/>`
Code:
```
 private MemoryStream stream;
 ...
 stream = new MemoryStream(new byte[width * height * 4]);
 ...
 private void OnPaint(byte[] pixelsBytes)
 {
     stream.Seek(0, SeekOrigin.Begin);
     stream.Write(pixelsBytes, 0, pixelsBytes.Length);
     ImageSource source = ImageSource.FromStream(() => stream);
     Image.Source = source;
 }
```
with no luck. ImageSource wants a stream from a graphics file, not a stream of a pixels byte array.
I also saw how to use GraphicsView with Canvas, however, it looks uncompleted. Microsoft.Maui.Graphics.Platform namespace is inaccessible on Windows and [there](https://docs.microsoft.com/en-us/dotnet/maui/user-interface/graphics/draw) are no examples for Windows.
Even if I could use:
```
 var image = PlatformImage.FromStream(stream);
 canvas.DrawImage(image, 10, 10, image.Width, image.Height);
```
MAUI awaits that stream is a stream of a graphics file from resources or disk, not a pixels array with channels of color. I think MAUI WebView does something similar to show web pages. Is there a way to draw pixel arrays dynamically on Image control or are there some other surfaces to do it? Could someone direct me, please?

**Comment by xtuzy** [comment]
use skiasharp or use platform method custom a control like https://github.com/xtuzy/MauiPlayground/tree/master/MauiLib/CustomControls to draw

**Comment by dxh-sc** [comment]
same question and also i can update WriteableBitmap content(with memory address) with hight fps.
example of video frame.

I'm confused if it has the same function on MAUI.
